koku/koku/trino_database.py

```diff
@@ -78,6 +84,130 @@ def query_id(self) -> t.Optional[str]:
         return self._trino_error.query_id
 
 
+class NoSuchKeyViolation:
```

Comment:
This class isn't adding much value. It can effectively be replaced with `"NoSuchKey" in str(ex)`.